### PR TITLE
Porechop

### PR DESCRIPTION
**Pull request by willemdek11**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/porechop/porechop.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<tool id="porechop" name="porechop" version="0.2.3">
+    <description>Adapter trimmer for Oxford Nanopore reads</description>
+    <requirements>
+        <requirement type="package" version="0.2.3_seqan2.1.1">porechop</requirement>
+    </requirements>
+    <version_command>porechop --version</version_command>
+    <command detect_errors="exit_code"><![CDATA[
+porechop
+    -i $input_file
```
Comment (bgruening) [inline tools/porechop/porechop.xml]: please use single quotes here

File: tools/porechop/porechop.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_file" ftype="fasta" value="test_format.fasta"/>
+            <param name="format" value="fasta"/>
+            <output name="outfile" ftype="fasta" file="out.fasta"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+Bandage is a GUI program that allows users to interact with the assembly graphs made by de novo assemblers such as Velvet, SPAdes, MEGAHIT and others.
```
Comment (bgruening) [inline tools/porechop/porechop.xml]: that seems to be the wrong help :)

File: tools/porechop/porechop.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+De novo assembly graphs contain not only assembled contigs but also the connections between those contigs, which were previously not easily accessible. Bandage visualises assembly graphs, with connections, using graph layout algorithms. Nodes in the drawn graph, which represent contigs, can be automatically labelled with their ID, length or depth. Users can interact with the graph by moving, labelling and colouring nodes. Sequence information can also be extracted directly from the graph viewer. By displaying connections between contigs, Bandage opens up new possibilities for analysing and improving de novo assemblies that are not possible by looking at contigs alone.
+    ]]></help>
+    <citations>
+        <citation type="doi">10.1093/bioinformatics/btv383</citation>
```
Comment (bgruening) [inline tools/porechop/porechop.xml]: wrong one

File: tools/porechop/porechop.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            <param argument="--extra_middle_trim_good_side" type="integer" min="0" value="10" optional="True" label="Extra trimming good side" help="This many additional bases will be removed next to middle adapters on their 'good' side (default: 10)"/>
+            <param argument="--extra_middle_trim_bad_side" type="integer" min="0" value="100" optional="True" label="Extra trimming bad side" help="This many additional bases will be removed next to middle adapters on their 'bad' side (default: 100)"/>
+            <param argument="--min_split_read_size" type="integer" min="0" value="1000" optional="True" label="Min. length reads post-split" help="Post-split read pieces smaller than this many base pairs will not be outputted (default: 1000)"/>
+        </section>
+    </inputs>
+    <outputs>
+        <data name="outfile" format="fasta" from_work_dir="out.*" label="${tool.name} on ${on_string}: Assembly Graph Image"/>
+    </outputs>
+    <tests>
```
Comment (bgruening) [inline tools/porechop/porechop.xml]: a few more tests, or a test with more parameters would be nice

File: tools/porechop/porechop.xml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            <param argument="--middle_threshold" type="float" min="0" max="100" value="85.0" optional="True" label="Min. percent identity" help="Adapters in the middle of reads must have at least this percent identity to be found (0 to 100) (default: 85.0)"/>
+            <param argument="--extra_middle_trim_good_side" type="integer" min="0" value="10" optional="True" label="Extra trimming good side" help="This many additional bases will be removed next to middle adapters on their 'good' side (default: 10)"/>
+            <param argument="--extra_middle_trim_bad_side" type="integer" min="0" value="100" optional="True" label="Extra trimming bad side" help="This many additional bases will be removed next to middle adapters on their 'bad' side (default: 100)"/>
+            <param argument="--min_split_read_size" type="integer" min="0" value="1000" optional="True" label="Min. length reads post-split" help="Post-split read pieces smaller than this many base pairs will not be outputted (default: 1000)"/>
+        </section>
+    </inputs>
+    <outputs>
+        <data name="outfile" format="fasta" from_work_dir="out.*" label="${tool.name} on ${on_string}: Assembly Graph Image"/>
```
Comment (shiltemann) [inline tools/porechop/porechop.xml]: the format isn't always `fasta` right? it depends on the selection of the `format` parameter. Could you change this and add a test for a different output format?

Comment (shiltemann) [reply tools/porechop/porechop.xml]: also, the output description should be updated :)

File: tools/porechop/porechop.xml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            <param argument="--discard_unassigned" type="boolean" checked="false" truevalue="--discard_unassigned" falsevalue="" label="Discard unassigned reads" help="Discard unassigned reads (instead of creating a 'none' bin) (default: False)"/>
+        </section>
+        <section name="adapter_search_settings" title="Adapter search settings">
+            <param argument="--adapter_threshold" type="float" min="0" max="100" value="90.0" optional="True" label="Min identity" help="An adapter set has to have at least this percent identity to be labelled as present and trimmed off (0 to 100) (default: 90.0)"/>
+            <param argument="--check_reads" type="integer" min="0" value="10000" optional="True" label="Number of alligned reads" help="This many reads will be aligned to all possible adapters to determine which adapter sets are present (default: 10000)"/>
+            <param argument="--scoring_scheme" type="text" value="3,-6,-5,-2" label="Comma-delimited string of alignment scores: match, mismatch, gap open, gap extend"/>
```
Comment (shiltemann) [inline tools/porechop/porechop.xml]: perhaps we would like to add a validator regex here as well, to ensure valid input?

**Comment by bgruening** [comment]
Thanks @willemdek11 and @shiltemann!

**Pull request merged**